### PR TITLE
Update roadmap 2023

### PR DESCRIPTION
**Pull request by mapedraza**

### CODE REVIEW

**Comment by AlvaroVega** [comment]
@mapedraza It would be great add also this issue https://github.com/telefonicaid/iotagent-node-lib/issues/1429 to iotagent roadmap.

**Comment by mapedraza** [comment]
This PR is ready to review

File: doc/roadmap.md [hunk 1]
```diff
@@ -26,32 +26,42 @@ The following list of features are planned to be addressed in the short term, an
...
     take)
--   Use the lightweight ingestion mechanism for connection oriented updates implemented in Context Broker
--   Add support to other transport protocols (BacNET, Modbus, etc)
 
 ### Features already completed
 
 The following list contains all features that were in the roadmap and have already been implemented.
 
+-   Refactor entities-NGSI-v2.js module (#1166) ([3.0.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/3.0.0))
+-   Accept JEXL Expressions for entity name in autoprovisioned devices (entityNameExp) (#1145) ([2.24.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.24.0))
```
Comment (fgalan) [inline doc/roadmap.md:63]: Issue number should be hyperlinks.

Comment (mapedraza) [reply doc/roadmap.md]: Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1419/commits/8148c7102498387bfa80b04d5822c9c43786f764

File: doc/roadmap.md [hunk 2]
```diff
@@ -26,32 +26,42 @@ The following list of features are planned to be addressed in the short term, an
...
 
 ### Medium term
 
 The following list of features are planned to be addressed in the medium term, typically within the subsequent
 release(s) generated in the next 9 months after the next planned release:
 
--   Accept JEXL Expressions for entity name in autoprovisioned devices (#1145)
--   Refactor entities-NGSI-v2.js module (#1166)
+-   Allow to add metadata to attributes from measures ([#1453](https://github.com/telefonicaid/iotagent-node-lib/issues/1453))
+-   Cache support
```
[the remark this answers is not in the document]
Comment (mapedraza) [reply doc/roadmap.md]: Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1419/commits/8148c7102498387bfa80b04d5822c9c43786f764

File: doc/roadmap.md [hunk 3]
```diff
@@ -26,40 +26,52 @@ The following list of features are planned to be addressed in the short term, an
...
 -   Incremental introduccion of ECMAScript6 syntax (previous analysis of which sub-set of interesting aspect we want to
     take)
--   Use the lightweight ingestion mechanism for connection oriented updates implemented in Context Broker
--   Add support to other transport protocols (BacNET, Modbus, etc)
 
 ### Features already completed
 
 The following list contains all features that were in the roadmap and have already been implemented.
 
+-   Refactor entities-NGSI-v2.js module ([#1166](https://github.com/telefonicaid/iotagent-node-lib/issues/1166)) ([3.0.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/3.0.0))
```
Comment (fgalan) [inline doc/roadmap.md:62]: Looking into release notes at https://github.com/telefonicaid/iotagent-node-lib/releases/tag/3.0.0, issue #1166 is not mentioned.

A bit weird, but I think it's ok anyway.

NTC

File: doc/roadmap.md [hunk 4]
```diff
@@ -26,40 +26,52 @@ The following list of features are planned to be addressed in the short term, an
...
+-   Accept JEXL Expressions for entity name in autoprovisioned devices (entityNameExp) ([#1145](https://github.com/telefonicaid/iotagent-node-lib/issues/1145)) ([2.24.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.24.0))
```
Comment (fgalan) [inline doc/roadmap.md:63]: I don't find #1145 mentioned in https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.24.0. However, it's mentioned in https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.22.0

Is this link ok?

File: doc/roadmap.md [hunk 5]
```diff
@@ -26,40 +26,52 @@ The following list of features are planned to be addressed in the short term, an
...
+-   Improve command functionalities (binary data + expression + mapping) ([2.22.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.22.0))
```
Comment (fgalan) [inline doc/roadmap.md:64]: No issue number associated in this case?

File: doc/roadmap.md [hunk 6]
```diff
@@ -26,40 +26,52 @@ The following list of features are planned to be addressed in the short term, an
...
 -   Support for "delta" measures (i.e. "temperature _increased_ in 5 degress" instead of "temperature _is_ 25")
 -   Allow to handle binary messages ([iota-ul#530](https://github.com/telefonicaid/iotagent-ul/issues/530))
 -   Removal support for NGSIv1 (#966) ([2.18.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.18.0))
```
Comment (fgalan) [inline doc/roadmap.md:67]:
```suggestion
-   Removal support for NGSIv1 ([#966]((https://github.com/telefonicaid/iotagent-node-lib/issues/966)) ([2.18.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.18.0))
```

File: doc/roadmap.md (hunk 4 above)
Comment (mapedraza) [reply doc/roadmap.md:63]: As far as I can see, merge commit of related PR is taged as 2.22.0:
https://github.com/telefonicaid/iotagent-node-lib/commit/ff6bfe9689c1423c9d78067eef5f51720975fa67

File: doc/roadmap.md (hunk 5 above)
Comment (mapedraza) [reply doc/roadmap.md:64]: No other related issues found in release notes apart from:
- Fix: apply expressions to pull commands (https://github.com/telefonicaid/iotagent-node-lib/issues/1263)

(also, 2.21.0 includes: Add: add expressions, payloadType and contentType to commands models)

File: doc/roadmap.md (hunk 3 above)
Comment (mapedraza) [reply doc/roadmap.md:62]: As far as I can see, merge commit of related PR is taged as 3.0.0:
https://github.com/telefonicaid/iotagent-node-lib/commit/9dc0657952ad1b9879a79afc8bd54134f9099957

File: doc/roadmap.md (hunk 4 above)
Comment (fgalan) [reply doc/roadmap.md:63]: > As far as I can see, merge commit of related PR is taged as 2.22.0: [ff6bfe9](https://github.com/telefonicaid/iotagent-node-lib/commit/ff6bfe9689c1423c9d78067eef5f51720975fa67)

That's confirm my point:

> However, it's mentioned in https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.22.0

So:

```suggestion
-   Accept JEXL Expressions for entity name in autoprovisioned devices (entityNameExp) ([#1145](https://github.com/telefonicaid/iotagent-node-lib/issues/1145)) ([2.22.0](https://github.com/telefonicaid/iotagent-node-lib/releases/tag/2.22.0))
```

It's correct?

File: doc/roadmap.md (hunk 5 above)
Comment (fgalan) [reply doc/roadmap.md:64]: Maybe existing

> Improve command functionalities (binary data + expression + mapping)

could be break into these two ones:

> Fix: apply expressions to pull commands
> Add: add expressions, payloadType and contentType to commands models

Each one assigned to its corresponding version (2.22.0 and 2.21.0 respectively)

Comment (fgalan) [reply doc/roadmap.md:64]: NTC

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**